packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -798,22 +820,29 @@ async fn directory_tree_to_loader_tree(
                 continue;
             }
 
-            if !tree.parallel_routes.contains_key("children") {
-                tree.parallel_routes.insert("children".to_string(), subtree);
-            } else {
-                // TODO: improve error message to have the full paths
-                DirectoryTreeIssue {
-                    app_dir,
-                    message: StyledString::Text(format!(
-                        "You cannot have two parallel pages that resolve to the same path. Route \
-                         {} has multiple matches in {}",
-                        for_app_path, app_page
-                    ))
-                    .cell(),
-                    severity: IssueSeverity::Error.cell(),
+            if let Some(current_tree) = tree.parallel_routes.get("children") {
+                if is_current_directory_catchall && *subtree.has_only_catchall().await? {
```

Comment:
```suggestion
                if *subtree.has_only_catchall().await? {
```
Or is there a reason this is necessary?

Reply:
A handful of tests fail if I remove this, but I'll need to dig in a bit more to understand why.